src/main/java/com/google/graphgeckos/dashboard/datatypes/BuildBotData.java

```diff
@@ -76,6 +80,59 @@ public BuildBotData(String commitHash, String name, List<Log> logs, BuilderStatu
     this.status = status;
   }
 
+  public BuildBotData(@JsonProperty("builderName") String name) {
+    this.name = name;
+  }
+
+  /**
+   * Extracts nested commitHash("revision"), branch("branch") and
+   * timestamp("when")fields from the json@param sourceStamp.
+   * Representation of the json component, where the commitHash field is located.
+   */
+  @JsonProperty("sourceStamp")
+  @SuppressWarnings("unchecked")
+  public void unpackSourceStamp(Map<String, Object> sourceStamp) {
+    commitHash = sourceStamp.get("revision").toString();
+    List<Object> changes = (ArrayList<Object>)sourceStamp.get("changes");
+    Map<String, Object> latestChange = (Map<String, Object>)changes.get(0);
+    timestamp = Timestamp.ofTimeMicroseconds(Long.parseLong(latestChange.get("when").toString()));
+  }
+
+  /**
+   * Defines builder status based on a phrase provided in parsed json.
+   * If something failed, then the buildbot is considered failed {@code FAILED}.
+   * If something is lost, then the buildbot is considered lost {@code LOST}.
+   * Otherwise the buildbot is considered passed {@code PASSED}.
```

Comment:
if `words` contains none of `{"failed", "lost", "successful"}`, the status is not updated. This comment hints that the buildbot is considered `PASSED` in this case

I'd suggest extracting a little helper function that returns the status given the list of `words` similar to the body of this one, that `return`-s where this one `break`-s and that returns `PASSED` by default, when none of the words matched `failed` or `lost`.

Reply:
Done, thanks!

Reply:
I don't see any changes to this comment nor to the `extractStatus` method below. Maybe the PR was not refreshed?